plutus-contract/src/Language/Plutus/Contract/Servant.hs

```diff
@@ -16,43 +16,29 @@ module Language.Plutus.Contract.Servant(
     , ContractResponse
     ) where
 
-import           Control.Monad.Except            (MonadError (..))
 import           Data.Aeson                      (FromJSON, ToJSON)
 import           Data.Proxy                      (Proxy (..))
 import           Data.Row
-import           Data.String                     (IsString (fromString))
-import           Servant                         ((:<|>) ((:<|>)), (:>), Get, JSON, Post, ReqBody, err500, errBody)
-import           Servant.Server                  (Application, Server, ServerError, serve)
+import           Servant                         ((:<|>) ((:<|>)), (:>), Get, JSON, Post, ReqBody)
+import           Servant.Server                  (Application, Server, serve)
 
 import           Language.Plutus.Contract.Schema (Event, Handlers, Input, Output)
 import           Language.Plutus.Contract.State  (ContractRequest, ContractResponse)
 import qualified Language.Plutus.Contract.State  as ContractState
 import           Language.Plutus.Contract.Types  (Contract)
 
-type ContractAPI s =
-       "initialise" :> Get '[JSON] (ContractResponse (Event s) (Handlers s))
-  :<|> "run" :> ReqBody '[JSON] (ContractRequest (Event s)) :> Post '[JSON] (ContractResponse (Event s) (Handlers s))
+type ContractAPI e s =
+       "initialise" :> Get '[JSON] (ContractResponse e (Event s) (Handlers s))
+  :<|> "run" :> ReqBody '[JSON] (ContractRequest (Event s)) :> Post '[JSON] (ContractResponse e (Event s) (Handlers s))
 
 -- | Serve a 'PlutusContract' via the contract API.
 contractServer
-    :: forall s e.
-       ( Show e
-       )
-    => Contract s e ()
-    -> Server (ContractAPI s)
+    :: forall s e
+    . Contract s e ()
+    -> Server (ContractAPI e s)
 contractServer con = initialise :<|> run where
-    initialise = servantResp (ContractState.initialiseContract con)
-    run req = servantResp (ContractState.insertAndUpdateContract con req)
-
-servantResp
-    :: (Show e, MonadError ServerError m)
-    => Either e (ContractResponse (Event s) (Handlers s))
-    -> m (ContractResponse (Event s) (Handlers s))
-servantResp = \case
-        Left err ->
-            let bd = "'insertAndUpdate' failed. " in
-            throwError $ err500 { errBody = fromString (bd <> show err) }
-        Right r -> pure r
+    initialise = pure $ ContractState.initialiseContract con
+    run req = pure $ ContractState.insertAndUpdateContract con req
```

Comment:
Do we not send errors back over HTTP now?

Reply:
Good point, I'll add it back in